File: api/Models/FullRecipe.cs
using System;
using System.Collections;
using System.Collections.Generic;
using api.Models;

namespace api.Models
{
    public class FullRecipe
    {
        //public string Name { get; set; }
        public int RecipeID { get; set; }
        public string Name { get; set; }
        public string ImagePath { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public float PrepTimeH { get; set; }
        public float PrepTimeM { get; set; }
        public float CookTimeH { get; set; }
        public float CookTimeM { get; set; }
        public List<Ingredient> Ingredients { get; set; }
        public List<Steps> Steps { get; set; }
        public List<SubSteps> SubSteps { get; set; }
        public List<Tips> Tips { get; set; }
        public List<SubTips> SubTips { get; set; }
#nullable enable
        public List<Photo>? AdditionalPhotos { get; set; }
        public string? PrepTime { get; set; }
        public string? CookTime { get; set; }
        public long Created { get; set; }
        public long LastModified { get; set; }

    }
}

File: Client/src/app/Models/FullRecipe.ts
import { Ingredient } from '../Entities/Ingredient';
import { Step } from '../Entities/Step';
import { Tip } from '../Entities/Tip';
import { AdditionalPhotos } from '../Entities/AdditionalPhotos';

export class FullRecipe {

    constructor(
        public RecipeID: number,
        public Name: string,
        public ImagePath: string,
        public Description: string,
        public Category: string,
        public PrepTimeH: number,
        public PrepTimeM: number,
        public CookTimeH: number,
        public CookTimeM: number,
        public Ingredients: Ingredient[],
        public Steps: Step[],
        public SubSteps: Step[],
        public Tips: Tip[],
        public SubTips: Tip[],
        public f: File[],
        // optional parameters
        public ExtraPhotos?: AdditionalPhotos[],
        public PrepTime?: string,
        public CookTime?: string,
        public Created?: number,
        public LastModified?: number,
        public ImageLoaded?: boolean
    ) { }
}
